core/src/main/java/com/github/jsonldjava/core/Context.java

```diff
@@ -187,7 +187,11 @@ private Context parse(Object localContext, List<String> remoteContexts,
             }
             // 3.2)
             else if (context instanceof String) {
-                String uri = (String) result.get(JsonLdConsts.BASE);
+                String uri = null;
+                // @base is ignored when processing remote contexts, https://github.com/jsonld-java/jsonld-java/issues/304
+                if (!context.toString().matches("^[hH][tT][tT][pP][sS]?://.*")) {
```

Comment:
What about having the regex compiled and somewhere static at the top of `Context.java`:

```java
private static final Pattern urlPattern = Pattern.compile("(?i)^https?://.*");
```
And then in that specific line do

```java
if (!urlPattern.matcher(context.toString()).matches())
```

This would be a bit more performant, since the Pattern does not need to be created for every call.